packages/polymorphic/src/Polymorphic.spec.tsx

```diff
@@ -0,0 +1,505 @@
+import React from 'react';
+import { parseTSDoc } from '../../../scripts/utils/tsDocParser';
+import { render } from '@testing-library/react';
+import styled from '@emotion/styled';
+import { Polymorph, usePolymorphicRef, type PolymorphicComponentType } from '.';
+import {
+  ExamplePolymorphic,
+  ExamplePolymorphicWithRef,
+  ExampleInferred,
+  AdvancedPolymorphic,
+  AdvancedPolymorphicWithRef,
+  RestrictedExample,
+} from './Example';
+import { makeWrapperComponent } from './utils/Polymorphic.testutils';
+
+describe('packages/polymorphic', () => {
+  describe('Basic Polymorphic Component', () => {
+    /* eslint-disable jest/no-disabled-tests */
+    test.skip('Prop Types behave correctly', () => {
+      const { Wrapper } = makeWrapperComponent();
+      const divRef = usePolymorphicRef<'div'>(); // React.useRef<HTMLDivElement | null>(null);
+      const anchorRef = usePolymorphicRef<'a'>();
+      const spanRef = usePolymorphicRef<'span'>();
+
+      <>
+        <Polymorph />
+        <Polymorph>some content</Polymorph>
+        <Polymorph as="div" />
+        <Polymorph as="div" ref={divRef} />
+        {/* @ts-expect-error - Must pass the correct ref type */}
+        <Polymorph as="div" ref={anchorRef} />
+        <Polymorph as="div" ref={divRef}>
+          some content
+        </Polymorph>
+        <Polymorph key="some-key" />
+        {/* @ts-expect-error href is not allowed on explicit div */}
+        <Polymorph as="div" href="mongodb.design" />
+
+        {/* @ts-expect-error - Require href when as="a" */}
+        <Polymorph as="a" />
+        <Polymorph as="a" href="mongodb.design" />
+        <Polymorph as="a" href="mongodb.design" ref={anchorRef} />
+        <Polymorph as="a" href="mongodb.design">
+          some content
+        </Polymorph>
+        {/* Should accept href (and render it as 'a') */}
+        {/* <Polymorph href="mongodb.design" /> */}
+
+        <Polymorph as="input" />
+        {/* TODO: ts-expect-error - Input should not accept children */}
+        {/* <Polymorph as="input">some content</Polymorph> */}
```

Comment:
is this a regression?

Reply:
No, TS doesn't even catch this for a straight `input`

Reply:
So we can rm